File: frontend/src/services/api/categoryService.js
import apiClient from './apiClient';

export const categoryService = {
  getAllCategories: async () => {
    const response = await apiClient.get('/categories');
    return response.data;
  },

  getCategoryById: async (id) => {
    const response = await apiClient.get(`/categories/${id}`);
    return response.data;
  },

  createCategory: async (data) => {
    const response = await apiClient.post('/categories', data);
    return response.data;
  },

  updateCategory: async (id, data) => {
    const response = await apiClient.put(`/categories/${id}`, data);
    return response.data;
  },

  deleteCategory: async (id) => {
    const response = await apiClient.delete(`/categories/${id}`);
    return response.data;
  },
};

File: backend/src/main/java/com/medistock/category/service/CategoryService.java
package com.medistock.category.service;

import com.medistock.category.dto.request.CategoryRequest;
import com.medistock.category.dto.response.CategoryDetailsResponse;
import com.medistock.category.dto.response.CategoryResponse;
import java.util.List;

public interface CategoryService {
    CategoryResponse createCategory(CategoryRequest request);
    List<CategoryResponse> getAllCategories();
    CategoryResponse getCategoryById(Long id);
    CategoryDetailsResponse getCategoryDetails(Long id);
    CategoryResponse updateCategory(Long id, CategoryRequest request);
    void deleteCategory(Long id);
}
